### DISCUSSION TITLE
Firewall@ and Message ID @firewall.something.....

### DISCUSSION THREAD

**Discussion by jassi0001** [martinschaible/rspamd-rules#39] [Ideas]
Hi Martin, 

thanks for your RSPAM Rules, very helpfull....

I have a question: I have a mailserver where i often ger Mails (SPAM) with the following content:

Message ID something like:
0.0.7.C25.1DA3A87DF189FFC.0@firewall.aycwebsolutions.com 

Sender IP different 

Sender firewall@aycwebsolutions.com

The TLD are different but allways the message id and the from look the same @firewall.domain.tld and from firewall@domain.tld.

Can you think abount a rule, which catches such messages... they are allways spam but you can not guess the messeage as spammy....

**Comment by martinschaible** [comment]
Hi,

this could be easy to catch by triggering several periods in the Message ID. This requires a new map.

**Comment by martinschaible** [comment]
I created a new map to catch the EMail address which was on my to do list. You asked for chatching the message id.
Let my check, how i can do this first.

**Comment by jassi0001** [comment]
Hi Martin, ich denke deutsch sollte auch klappen ;-)

Die Mails, die mit der Message ID kommen benötigen lediglich einen Abgleich zwischen dem, was hinter der Message ID steht und der Mailadresse. Die Maisl sind vermutlich auch "Rezept-Frei" Mails, jeoch ist es immer so, dass das, was hinter dem @ der Message ID steht immer identisch mit der Absende-Adresse ist. Das passiert bei "normalen" Mails eigentlich nie, dass es so eine Kombination gibt.
Vielleicht hilft das ja beim Versändnis. 

Ich habe es gerade mal ohne weitere Regeln ausprobiert. Die ganzen Mails werden auf Grund deiner Regeln bereits wegen "Rezeptfrei bestellen" erkannr und entsprechend gefiltert.... Evtl. ist es daher gar nicht notwendig einen Kombifilter mit Message ID und From Adresse zu bauen....

PS: Ich finde dein Projhekt wirklich großartig und es hilft wirklich gut weiter.
Wenn ich hier was testen soll, kann ich das gern machen. Momentan habe ich die IP Adressen von wo sowas kommt, ausgesperrt...

**Comment by martinschaible** [comment]
Hallo

Die Message ID alleine kann ich nicht nutzen. Sie eignet sich nicht für Spam-Filtering. Wenn also wie du schreibst ein Vergleich der Domain von FROM und MessageID helfen kann, wäre da ein LUA Script das Richtige. Da hatte ich bis einfach keine Zeit mit damit zu befassen. LUA selber ist kein Problem, da die Script-Sprache auf Erstklässler-Niveau ist. Aber im Zusammenhang mit Rspamd wird es dann sehr schnell undurchsichtig. Erstens ist die Doku mau und die "Community" in Telegram für ist einfachere Fragen absolut nicht zu haben. 

Du kannst mir aber durchgeschlüpften Spam an eine spezielle Mail Adresse weiterleiten. Diese möchte ich aber hier nicht zeigen. 
Solche Mails werden dann angeschaut und ich mache passende Regeln.
